main.rb

```diff
@@ -0,0 +1,186 @@
+    new_self.length.times do
+      yield(new_self[i]) == true ? arr = arr.push(new_self[i]) : arr
+      i += 1
+    end
+    arr
+  end
+
+  def my_all?(parameter = nil)
+    i = 0
+    check = 0
+    new_self = self
+    new_self = new_self.instance_of?(Array) || new_self.instance_of?(Hash) ? flatten : to_a
+    while block_given? && i < new_self.length
+      yield(new_self[i]) == true ? check += 1 : check = check
+      i += 1
+    end
+    if parameter.instance_of?(Regexp)
+      new_self.length.times do
+        new_self[i] =~ parameter ? check += 1 : check = check
+        i += 1
+      end
+    else
+      new_self.length.times do
+        new_self[i].class == parameter ? check += 1 : check = check
+        i += 1
+      end
+    end
+    check == new_self.length
+  end
+
+  def my_any?(parameter = nil)
+    i = 0
+    check = 0
+    new_self = self
+    new_self = new_self.instance_of?(Array) || new_self.instance_of?(Hash) ? flatten : to_a
+    while block_given? && i < new_self.length
+      yield(new_self[i]) == true ? check += 1 : check = check
+      i += 1
+    end
+    if parameter.instance_of?(Regexp)
+      new_self.length.times do
+        new_self[i] =~ parameter ? check += 1 : check = check
+        i += 1
+      end
+    else
+      new_self.length.times do
+        new_self[i].class == parameter ? check += 1 : check = check
+        i += 1
+      end
+    end
+    check.positive? ? true : false
+  end
+
+  def my_none?(parameter = nil)
+    i = 0
+    check = 0
+    new_self = self
+    new_self = new_self.instance_of?(Array) || new_self.instance_of?(Hash) ? flatten : to_a
+    while block_given? && i < new_self.length
+      yield(new_self[i]) == true ? check += 1 : check = check
+      i += 1
+    end
+    if parameter.instance_of?(Regexp)
+      new_self.length.times do
+        new_self[i] =~ parameter ? check += 1 : check = check
+        i += 1
+      end
+    else
+      new_self.length.times do
+        new_self[i].class == parameter ? check += 1 : check = check
+        i += 1
+      end
+    end
+    check.positive? ? false : true
+  end
+
+  def my_count
+    return enum_for(:my_count) unless block_given?
+
+    self_length = length
+    i = 0
+    new_self = self
+    new_self = new_self.instance_of?(Array) || new_self.instance_of?(Hash) ? flatten : to_a
+    length.times do
+      yield(new_self[i]) == false ? self_length -= 1 : self_length = self_length
+      i += 1
+    end
+    self_length
+  end
+
+  def my_map(my_proc = nil)
+    arr = []
+    i = 0
+    new_self = self
+    new_self = new_self.instance_of?(Array) || new_self.instance_of?(Hash) ? flatten : to_a
+    if block_given?
+      new_self.length.times do
+        arr = arr << yield(new_self[i])
+        i += 1
+      end
+    elsif !my_proc.nil?
+      new_self.length.times do
+        arr = arr << my_proc.call(new_self[i])
+        i += 1
+      end
+    else
+      return enum_for(:my_map)
+    end
+    arr
+  end
```

Comment:
# my_map
- [ ] `my_map` executes only the proc when both a block and a proc are given and returns a new array with the results of running the proc once for every element in enum.

      Failure/Error: expect(array.my_map(my_proc) { |num| num < 10 }).to eq array.map(&my_proc)

        expected: [false, false, false, false, false, false, false, false, false, false, false, false, false, false, fa...e, false, false, false, false, false, false, false, false, false, false, false, false, false, false]
             got: [true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true...rue, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true]